### PR TITLE
TF-564: Add Prettier and ESLint to Web Settings

### PR DESCRIPTION
**Pull request by JordanBerni-UL**
- [ ] Code Review
- [ ] Developer Testing: Key item tests

### CODE REVIEW

File: TF_Settings_Web/.eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+            }
+        ],
+        "@typescript-eslint/no-empty-interface": "warn",
+        "unicorn/prevent-abbreviations": "off",
+        "unicorn/filename-case": "off",
+        "unicorn/numeric-separators-style": "off",
+        "unicorn/no-null": "off",
+        "unicorn/no-zero-fractions": "off",
+        "unicorn/no-array-push-push": "off",
+        "unicorn/no-nested-ternary": "off"
```
Comment (GabrielHomewoodUL) [inline TF_Settings_Web/.eslintrc.json:51]: I think we should maybe increase the max line length from the default as by default it is 80 characters which is a bit short!
e.g.
`"max-len": 120`

File: TF_Settings_Web/src/Components/Pages/InteractionsPage.tsx [hunk 2]
```diff
@@ -68,17 +67,17 @@ export class InteractionsPage extends Page<{}, interactionsState> {
             console.error(`Could not change interaction type; did not recognise the "${newValue}" interaction`);
         }
 
-        let interactionType: InteractionType = InteractionTranslator[newValue];
+        const interactionType: InteractionType = InteractionTranslator[newValue];
```
Comment (NatGreenwood-UL) [inline TF_Settings_Web/src/Components/Pages/InteractionsPage.tsx:70]: Didn't realise `const` was block-scoped the same as `let`! Good to know 👍

File: TF_Settings_Web/src/Components/Pages/Page.tsx [hunk 3]
```diff
@@ -1,22 +1,15 @@
...
-    render () {
-        return (
-            <div>
-                This is the base Page Render.
-                You should not be seeing this.
-            </div>
-        );
+import { Component } from 'react';
+
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
```
Comment (NatGreenwood-UL) [inline TF_Settings_Web/src/Components/Pages/Page.tsx]: is there a way to disable things like this per-file to avoid repeated comment lines of eslint?

Comment (JordanBerni-UL) [reply TF_Settings_Web/src/Components/Pages/Page.tsx]: Yes there is - adding  `/* eslint-disable @typescript-eslint/no-empty-interface */` at the top of the file will do that

Comment (GabrielHomewoodUL) [reply TF_Settings_Web/src/Components/Pages/Page.tsx]: You can do something like:

```
/* eslint-disable no-alert, no-console */

alert('foo');
console.log('bar');

/* eslint-enable no-alert, no-console */
```

**Comment by NatGreenwood-UL** [comment]
Only thing I'd like to see before this is merged is something added to the root-level readme (or the project-level readme honestly) detailing how to set this up and run the linter locally. I know that I'm not going to remember to include some of the more minor changes (lowercase hex codes, leading 0s) so I'd need to run it myself, and documenting it now saves us having to figure out / explain it again as new members join the team. 👍

**Comment by JordanBerni-UL** [comment]
I've updated the readme @NatGreenwood-UL @GabrielHomewoodUL

**Review by NatGreenwood-UL — APPROVED** [review]

**Pull request merged**